src/samplers/memory/proc_meminfo/mod.rs

```diff
@@ -0,0 +1,100 @@
+use crate::common::LazyGauge;
+use metriken::MetricInstance;
+
+use super::stats::*;
+use super::*;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{Read, Seek};
+
+#[cfg(target_os = "linux")]
+#[distributed_slice(MEMORY_SAMPLERS)]
+fn init(config: &Config) -> Box<dyn Sampler> {
+    Box::new(ProcMeminfo::new(config))
+}
+
+pub struct ProcMeminfo {
+    prev: Instant,
+    next: Instant,
+    interval: Duration,
+    file: File,
+    gauges: HashMap<&'static str, &'static MetricInstance<LazyGauge>>,
+}
+
+impl ProcMeminfo {
+    #![allow(dead_code)]
+    pub fn new(_config: &Config) -> Self {
+        let now = Instant::now();
+
+        let gauges = HashMap::from([
+            ("MemTotal:", &MEMORY_TOTAL),
+            ("MemFree:", &MEMORY_FREE),
+            ("MemAvailable:", &MEMORY_AVAILABLE),
+            ("Buffers:", &MEMORY_BUFFERS),
+            ("Cached:", &MEMORY_CACHED),
+        ]);
+
+        Self {
+            file: File::open("/proc/meminfo").expect("file not found"),
```

Comment:
Similar to `Hwinfo`, I recommend defining data source for each sampler as a constant at the module level.